runtime/near-vm-runner/src/runner.rs

```diff
@@ -27,7 +27,39 @@ pub fn run<'a>(
     promise_results: &'a [PromiseResult],
     current_protocol_version: ProtocolVersion,
     cache: Option<&'a dyn CompiledContractCache>,
+    #[allow(unused_variables)] profile: &Option<ProfileData>,
 ) -> (Option<VMOutcome>, Option<VMError>) {
+    #[cfg(feature = "costs_counting")]
```

Comment:
if we do this why do we not put the function argument behind a feature flag so that we don't have to do something like `#[allow(unused_variables)]`

Reply:
Because the caller always passes the parameter since it's in `ApplyState` and by default is None, but we only use it when the feature is on. 

Making the parameter be behind the feature would require moving each call behind a feature flag too.  Is this what you want?

Reply:
It sounds like a good idea. For example on-disk compilation cache introduced a few unwanted changes. Hiding a feature behind a flag makes it safer for non-prod changes.